### ISSUE TITLE
Problem including the package dependency in Maven

### ISSUE THREAD

**Issue by axymthr** [xebia-os/xebia-archunit-extension#1]
Getting the following error on adding into pom.xml
```
    <repositories>
        <repository>
            <id>github</id>
            <name>GitHub Packages</name>
            <url>https://maven.pkg.github.com/xebia-os/xebia-archunit-extension</url>
        </repository>
    </repositories>

        <dependency>
            <groupId>com.github.xebia</groupId>
            <artifactId>archunit-extension</artifactId>
            <version>0.1.0</version>
            <scope>test</scope>
        </dependency>
```

> Could not find artifact com.github.xebia:xebia-archunit-extension:pom:0.1.0 in github (https://maven.pkg.github.com/xebia-os/xebia-archunit-extension)

> Could not resolve dependencies for project com.xebia.starters:api:jar:0.1.0-SNAPSHOT: Failed to collect dependencies at com.github.xebia:archunit-extension:jar:0.1.0: Failed to read artifact descriptor for com.github.xebia:archunit-extension:jar:0.1.0: Failure to find com.github.xebia:xebia-archunit-extension:pom:0.1.0 in https://maven.pkg.github.com/xebia-os/xebia-archunit-extension was cached in the local repository, resolution will not be reattempted until the update interval of github has elapsed or updates are forced 

Already created a Github PAT and authenticated to Github Packages. The URL seems to be right as per the Github docs but I see a 404 on that page. I can download the jar manually from the packages page. What is missing? Is the name a problem? archunit-extension vs xebia-archunit-extension?

**Comment by shekhargulati** [comment]
Follow this https://docs.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-apache-maven-for-use-with-github-packages#authenticating-to-github-packages

**Comment by axymthr** [comment]
Get the following message now:

> Failed to collect dependencies at com.github.xebia:archunit-extension:jar:0.1.0: Failed to read artifact descriptor for com.github.xebia:archunit-extension:jar:0.1.0: Could not find artifact com.github.xebia:xebia-archunit-extension:pom:0.1.0 in github (https://maven.pkg.github.com/xebia-os/xebia-archunit-extension) 

My settings.xml
```
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0
                      http://maven.apache.org/xsd/settings-1.0.0.xsd">

    <activeProfiles>
        <activeProfile>github</activeProfile>
    </activeProfiles>

    <profiles>
        <profile>
            <id>github</id>
            <repositories>
                <repository>
                    <id>github</id>
                    <name>GitHub OWNER Apache Maven Packages</name>
                    <url>https://maven.pkg.github.com/xebia-os/xebia-archunit-extension</url>
                </repository>
                <repository>
                    <id>central</id>
                    <url>https://repo1.maven.org/maven2</url>
                    <releases><enabled>true</enabled></releases>
                    <snapshots><enabled>true</enabled></snapshots>
                </repository>
            </repositories>
        </profile>
    </profiles>

    <servers>
        <server>
            <id>github</id>
            <username>OMITTED HERE</username>
            <password>OMITTED HERE</password>
        </server>
    </servers>
</settings>
```

Dependency added to pom.xml
```
        <dependency>
            <groupId>com.github.xebia</groupId>
            <artifactId>archunit-extension</artifactId>
            <version>0.1.0</version>
            <scope>test</scope>
        </dependency>
```

Removed the <repository> tag from pom.xml this time since it is now in settings.xml
<img width="488" alt="Screenshot 2020-09-07 at 2 39 28 PM" src="https://user-images.githubusercontent.com/17650429/92370471-f4185a80-f117-11ea-8aec-e961c7fd3775.png">
For some reason IntelliJ is looking for 2 different dependencies?

**Comment by axymthr** [comment]
Maven is kind of iffy to work with anyhow, let me try with a Gradle build.

**Comment by shekhargulati** [comment]
Have you added repository in the pom.xml as well?

**Comment by axymthr** [comment]
### Gradle test
1. Generated a new Spring Web project from start.spring.io

2. Ran `./gradlew build`

3. Followed https://docs.github.com/en/packages/using-github-packages-with-your-projects-ecosystem/configuring-gradle-for-use-with-github-packages#authenticating-to-github-packages to add configuration to build.gradle. The Github docs are kind of iffy/unclear, ended up with the following:
```
plugins {

	id("maven-publish")
}
repositories {

	maven {
		name = "GitHubPackages"
		url = uri("https://maven.pkg.github.com/xebia-os/archunit-extension")
		credentials {
			username = OMITTED
			password = OMITTED
		}
	}
}
dependencies {

	testImplementation("com.github.xebia:archunit-extension:0.1.0")
}
```
Got the following error:
```
./gradlew build
> Task :compileTestJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileTestJava'.
> Could not resolve all files for configuration ':testCompileClasspath'.
   > Could not resolve com.github.xebia:archunit-extension:0.1.0.
     Required by:
         project :
      > Could not resolve com.github.xebia:archunit-extension:0.1.0.
         > Could not parse POM https://maven.pkg.github.com/xebia-os/archunit-extension/com/github/xebia/archunit-extension/0.1.0/archunit-extension-0.1.0.pom
            > Could not find com.github.xebia:xebia-archunit-extension:0.1.0.

```

**Comment by axymthr** [comment]
> Have you added repository in the pom.xml as well?

Yes I did initially. Tried with both.

**Comment by axymthr** [comment]
One confusion, is it archunit-extension or xebia-archunit-extension?

**Comment by shekhargulati** [comment]
https://github.com/xebia-os/xebia-archunit-extension/packages/374835

**Comment by shekhargulati** [comment]
The problem was that parent pom was not published. I fixed the publish process in version `0.2.0`. @awkshwayrd  please close this if it is working for you now.

**Comment by axymthr** [comment]
Working for me now so closing it.
